### ISSUE TITLE
[UI] What exactly does "background sync" mean?

### ISSUE THREAD

**Issue by hrj** [vector-im/riot-android#2044]
While discussing in the `riot-android` room, `Coffee` wrote:

> enabling background sync will strip the messages going through google's push server. The push will be nothing but a prod to get Riot to fetch the message from the homeserver directly.
> Otherwise, message contents are sent through the google push server.

If this is true, then the UI needs much more clarity regarding the following:

1. What does "background sync" do when enabled?
2. Is this setting only applicable when the application is not-in-the-foreground?
3. What happens when this setting is disabled? Does it just disable the functionality in (1) or it also enables a different type of functionality?
4. Does this setting affect the number of third-parties involved? If third-parties are involved because of this setting, then I would like to see a verbose description for the setting.

thanks!

**Comment by manuroe** [comment]
There is now a doc explaining how notifications works on Riot: [notifications.md](https://github.com/vector-im/riot-android/blob/e0e703d09a8b330d2779e75485c0ad0181d7452a/docs/notifications.md).

**Comment by hrj** [comment]
@manuroe Thanks. It clarifies matters a lot.

Read the doc, it appears as if the "background sync" setting is overloaded. It means different things in different contexts. I see that in #2051 you suggest creating a new setting for the type of data that is sent to GCM. I agree, and, in fact, I would take it further: have even more settings that spell everything out (or settings with more options, not just a toggle).

For example, one idea could be to use a setting with 3 radio buttons:
```
Background Sync
  * Disable
  * Poll with homeserver
  * Use Google's push service (This option can be hidden in fdroid build)
```

If the third option is enabled, then the user could be shown another setting for what gets pushed to third-party (as you suggest in #2051).

**Comment by hrj** [comment]
There seems to be a link between rage shake and background sync. In `vector/src/main/java/im/vector/fragments/VectorSettingsPreferencesFragment.java` line 850, there is this code:

```
        if(mIsUsedRageShake) {
            useRageShakeModePref.setChecked(true);
        } else {
            useBackgroundSyncPref.setChecked(false);
        }
```
I am not sure how these are related!

**Comment by manuroe** [comment]
> I am not sure how these are related!
@hrj , there are absolutely not related. You discover allows to fix #2075.
Thanks a lot.

**Comment by manuroe** [comment]
I close the issue because we have now a doc describing how data are shared with GCM ([notifications.md](https://github.com/vector-im/riot-android/blob/master/docs/notifications.md)) and we have added a new setting to not send message content to GCM without the user's consent (https://github.com/vector-im/riot-android/pull/2059).
